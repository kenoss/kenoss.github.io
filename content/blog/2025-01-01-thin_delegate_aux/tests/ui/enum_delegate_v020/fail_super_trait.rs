use enum_delegate_v020 as enum_delegate;

#[enum_delegate::register]
trait ShapeI: std::fmt::Debug {
    fn area(&self) -> f64;
}

#[derive(Debug)]
struct Rect {
    width: f64,
    height: f64,
}

#[derive(Debug)]
#[enum_delegate::implement(ShapeI)]
enum Shape {
    Rect(Rect),
}

impl ShapeI for Rect {
    fn area(&self) -> f64 {
        self.width * self.height
    }
}

fn main() {}
