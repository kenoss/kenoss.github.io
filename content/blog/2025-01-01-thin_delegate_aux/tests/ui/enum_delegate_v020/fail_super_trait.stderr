error: enum_delegate does not currently support supertraits
  --> tests/ui/enum_delegate_v020/fail_super_trait.rs:4:15
   |
4  | trait ShapeI: std::fmt::Debug {
   |               ^^^
...
15 | #[enum_delegate::implement(ShapeI)]
   | ----------------------------------- in this procedural macro expansion
   |
   = note: this error originates in the macro `ShapeI` which comes from the expansion of the attribute macro `enum_delegate::implement` (in Nightly builds, run with -Z macro-backtrace for more info)
